src/java/org/apache/cassandra/service/StorageService.java

```diff
@@ -114,6 +115,7 @@ public class StorageService extends NotificationBroadcasterSupport implements IE
     private int cleanupOpsInProgress = 0;
 
     private final RepairTracker repairTracker = new RepairTracker();
+    private final List<ProgressListener> bootstrapListeners = new CopyOnWriteArrayList<>();
```

Comment:
why use a CopyOnWriteArrayList?

Reply:
thread-safe + follows existing usage for `lifecycleSubscribers`